### PR TITLE
Respect my..theme

### PR DESCRIPTION
**Pull request by Eonasdan**
Hi. Thanks for the great repo.

I want to slowly help to make the library fully respect theme settings so that a dark theme is possible. This is related to #82 and #470. 

I fixed the drawer and the body background/color. If you want to check it you can use theme I'm trying to get to work:

```
MatTheme Theme = new MatTheme()
  {
    Primary = "#1976D2",
    Secondary = "#9C27B0",
    Background = "#303030",
    Surface = "#333333",
    OnPrimary = "#fff",
    OnSecondary = "#201317",
    OnSurface = "#fff"
  };
```

image

The commented out code is because the angular material docs drawer has some extra styles that I was trying to get to work to make the items stand out a little more but couldn't get it to work just right in the time I had.

### CODE REVIEW

File: src/MatBlazor/Components/MatCard/MatCardContent.razor [hunk 1]
```diff
@@ -18,7 +18,6 @@
          protected override async Task OnInitializedAsync()
          {
              await base.OnInitializedAsync();
-             ClassMapper.Add("mdc-card__primary-action");
```
Comment (lindespang) [inline src/MatBlazor/Components/MatCard/MatCardContent.razor:21]: @Eonasdan Did you try to pull a little sneaky here? ;- ) (Saw your Issue but let's take discussion on this there instead and make this a separate PR)

Comment (Eonasdan) [reply src/MatBlazor/Components/MatCard/MatCardContent.razor:21]: Not sneaky lol. It was just causing issues in my application. I remove this line from the pr and do branches in my own pr to keep stuff separate.

Comment (Eonasdan) [reply src/MatBlazor/Components/MatCard/MatCardContent.razor:21]: done

**Comment by SamProf** [comment]
I think you make really good start, but it's very unsafe to change `:root` styles

**Comment by Eonasdan** [comment]
> I think you make really good start, but it's very unsafe to change `:root` styles

@SamProf I don't really seen another way. Unless we could pull in the sass files and override those straight from material. There's a couple of places where having that change was the only way to apply the theme.

**Comment by cloudb0x** [comment]
Whats holding us up here? I'm having to manually apply the theme around my app. Horrible experience anything I can do to help?

**Comment by mawendel** [comment]
I'd like to chime in here and ask if there will be any movement on this? Dark mode is pretty much a requirement for all new webapps in 2020 :-)

**Comment by Eonasdan** [comment]
the project I was demoing with Blazor died so I don't have a lot of motivation to make more progress. I also didn't get any additional feedback on changes I was trying to make. I'm willing to work on the conflicts if @SamProf still wants this pr

**Comment by BlowaXD** [comment]
Hi,
Thanks for your work on that library @SamProf 
Any news about the theme customization (dark theme more specifically) ? 
I just landed on Blazor for a PoC in my company, I used your library and the dark theme would fit more with our graphical charter and my eyes would thank you for saving some of their lifespan 👀 

I'm not used to frontend things at all and I don't really have the time to invest on that simple PoC

Thanks for considering my request

**Comment by Christian-Oleson** [comment]
For those asking about this, I believe someone with the time and resources will need to find a safer way to make this PR possible. I know myself and at least 1 other contributed have been unable to find time with very young kids, so any help getting this one to the finish line is appreciated.

**Comment by Christian-Oleson** [comment]
@Eonasdan ,
There are merge conflicts here

**Comment by Eonasdan** [comment]
yup but it sounded like you didn't want to merge the changes: "will need to find a safer way to make this PR possible" and " but it's very unsafe to change :root styles".

I can try to fix the conflicts but not if it won't be accepted anyway

**Pull request closed without merging**